src/wrappers/google/pubsub/SubscriptionExecutor.ts

```diff
@@ -0,0 +1,112 @@
+import Utils from '../../../utils/Utils';
+import {
+    ClassNames,
+    DomainNames,
+    TriggerHeaderTags,
+    SpanTags,
+    SpanTypes,
+    GooglePubSubTags,
+} from '../../../Constants';
+import PluginContext from '../../../plugins/PluginContext';
+import ExecutionContext from '../../../context/ExecutionContext';
+import WrapperUtils from '../../WebWrapperUtils';
+import InvocationSupport from '../../../plugins/support/InvocationSupport';
+import * as opentracing from 'opentracing';
+import ThundraSpanContext from '../../../opentracing/SpanContext';
+import InvocationTraceSupport from '../../../plugins/support/InvocationTraceSupport';
+import GooglePubSubUtils from '../../../utils/GooglePubSubUtils';
+import ConfigNames from '../../../config/ConfigNames';
+import ConfigProvider from '../../../config/ConfigProvider';
+
+const get = require('lodash.get');
+
+/**
+ * Start invocation process
+ * @param {PluginContext} pluginContext
+ * @param {ExecutionContext} execContext
+ */
+export function startInvocation(pluginContext: PluginContext, execContext: ExecutionContext) {
+    execContext.invocationData = WrapperUtils.createInvocationData(execContext, pluginContext);
+}
+
+/**
+ * Finish invocation process
+ * @param {PluginContext} pluginContext
+ * @param {ExecutionContext} execContext
+ */
+export function finishInvocation(pluginContext: PluginContext, execContext: ExecutionContext) {
+    WrapperUtils.finishInvocationData(execContext, pluginContext);
+}
+
+/**
+ * Start trace
+ * @param {PluginContext} pluginContext
+ * @param {ExecutionContext} execContext
+ */
+export function startTrace(pluginContext: PluginContext, execContext: ExecutionContext) {
+
+    const {
+        tracer,
+        request,
+    } = execContext;
+
+    const contextInformation: any = execContext.getContextInformation();
+
+    const propagatedSpanContext: ThundraSpanContext =
+        tracer.extract(opentracing.FORMAT_TEXT_MAP, request.attributes) as ThundraSpanContext;
+
+    const triggerOperationName = get(request, `attributes.${TriggerHeaderTags.RESOURCE_NAME}`)
```

Comment:
What if publisher side is not instrumented with Thundra agent so there is no resource name in the attributes and no operation name in the context. What will do in this case?

For that case what about setting trigger operation name (and the span operation name) to the subscription name:
```js
const subscriptionName = (subscription.metadata && subscription.metadata.topic)
        ? subscription.metadata.topic : subscription.name;
```